plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/MultiParentCompileHook.java

```diff
@@ -114,22 +117,51 @@ private void copy(Path eslintrc, File pluginFolder) {
         return mconfig;
     }
 
-    private void compile(MavenRunner.Config mavenConfig, File path, String parentFolder, String pluginName) throws PomExecutionException, IOException {
-        if (StringUtils.isBlank(parentFolder)) {
-            runner.run(mavenConfig, path, setupCompileResources(path), "clean", "process-test-classes", "-Dmaven.javadoc.skip");
-        } else {
-            if (!path.getAbsolutePath().contains(parentFolder)) {
-                throw new IOException(String.format("Something is really wrong here! parentFolder:%s not present in path %s", parentFolder, path.getAbsolutePath()));
-            }
-            if (!StringUtils.equals(parentFolder, path.getParentFile().getName())) {
-                throw new IOException(String.format("Something is really wrong here! %s is not the parent folder of %s", parentFolder, path.getAbsolutePath()));
-            }
+    private void compile(MavenRunner.Config mavenConfig, File path, File localCheckoutDir, String parentFolder, String pluginName) throws PomExecutionException, IOException {
+        if (isSnapshotJep229EnabledMultiParentPlugin(parentFolder, path, localCheckoutDir)) {
             // "process-test-classes" not working properly on multi-module plugin. See https://issues.jenkins.io/browse/JENKINS-62658
             // using "mvn clean install -DskipTests -Dinvoker.skip -Denforcer.skip to skip testing and just compile and install dependencies"
             runner.run(mavenConfig, path.getParentFile(), setupCompileResources(path.getParentFile()), "clean", "install", "-DskipTests", "-Dinvoker.skip", "-Denforcer.skip", "-Dmaven.javadoc.skip", "-am", "-pl", pluginName);
+        } else {
+            runner.run(mavenConfig, path, setupCompileResources(path), "clean", "process-test-classes", "-Dmaven.javadoc.skip");
         }
     }
 
+    /**
+     * Checks if a plugin is a multiparent plugin with JEP-229 enabled (see: https://github.com/jenkinsci/jep/blob/master/jep/229/README.adoc)
+     * and an SNAPSHOT project.version and without local checkout directory overriden. 
```

Comment:
The code does what it should be doing, I think, but the method name and Javadoc do not match.

All you need here is to verify that the checkout is of a snapshot. The main way this might happen is in fact that JEP-229 is in use and the commit we are cloning happens to have been promoted as a release. But there might be other ways building a multimodule project could otherwise be broken in a CI system with an empty local repo:

* a regular MRP plugin but overridden with a local clone of a branch or some other non-tag revision
* an incrementalified (JEP-305), but MRP-based not CD-enabled (JEP-229) plugin, checked out at a non-tagged commit (e.g. to test something in a draft PR in `jenkinsci/bom`)
* a non-incrementalified plugin checked out at an arbitrary commit, if PCT in the future supports timestamped snapshots (BEE-1751 in our internal tracker; not very interesting in @jenkinsci since we would just use Incrementals instead)